lib/ansible/utils/path.py

```diff
@@ -103,3 +104,9 @@ def basedir(source):
         dname = os.path.abspath(dname)
 
     return to_text(dname, errors='surrogate_or_strict')
+
+
+def cleanup_tmp_dir(path):
+    """Removes directory"""
+    if os.path.isdir(path):
+        shutil.rmtree(path)
```

Comment:
I'm not sure we want to traceback on a failure to remove the tmpdir.  We should probably wrap this in a try/except, and either explicitly ignore the exception, or warn.

Reply:
ignore if it is missing, warn if we fail to remove